op-batcher/batcher/driver.go

```diff
@@ -174,9 +173,18 @@ func (l *BatchSubmitter) Stop() error {
 	}
 	l.running = false
 
-	l.cancel()
-	close(l.done)
+	// go routine will call cancelKill() if the passed in ctx is ever Done
+	cancelKill := l.cancelKillCtx
+	wrapped, cancel := context.WithCancel(ctx)
+	defer cancel()
+	go func() {
+		<-wrapped.Done()
+		cancelKill()
+	}()
+
+	l.cancelShutdownCtx()
 	l.wg.Wait()
+	l.cancelKillCtx()
```

Comment:
```suggestion
	l.cancelShutdownCtx()
	defer l.cancelKillCtx()
	l.wg.Wait()

```
I think we fixed that in an earlier iteration already.

Reply:
This was intentionally re-introduced in order to allow us to use the passed-in context as a potential "kill" signal on shutdown: https://github.com/ethereum-optimism/optimism/pull/5105#discussion_r1149618650

Reply:
I see, sorry. Can't we then just do?:
```suggestion
	// go routine will call cancelKill() if the passed in ctx is ever Done
	go func() {
		<-ctx.Done()
		l.cancelKillCtx()
	}()

	l.cancelShutdownCtx()
	l.wg.Wait()
	l.cancelKillCtx()
```
What's the purpose of `wrapped`?

Reply:
Without `wrapped` and, more-importantly, the `cancel` signal, it could be possible to have the spawned goroutine run forever if the passed in `ctx` is never "done", like for example in the case where `ctx` is `context.Background()`

Reply:
Persisting a comment from private DM here: could also have done the following to avoid creating another context:
```suggestion
	go func() {
		select: {
		case <-ctx.Done():
			l.cancelKillCtx()
		// make sure goroutine returns eventually
		case <-l.killCtx:
			return
		}
	}()

	l.cancelShutdownCtx()
	l.wg.Wait()
	l.cancelKillCtx()
```